src/librustc/hir/map/mod.rs

```diff
@@ -1093,6 +1095,7 @@ impl<'a> print::State<'a> {
             Node::AnonConst(a)    => self.print_anon_const(&a),
             Node::Expr(a)         => self.print_expr(&a),
             Node::Stmt(a)         => self.print_stmt(&a),
+            Node::PathSegment(_)  => bug!("cannot print PathSegment"),
```

Comment:
You should probably implement this, to avoid cascading issues.

Reply:
done